app/facade/partner_facade.rb

```diff
@@ -0,0 +1,37 @@
+class PartnerFacade
+
+  attr_reader :partner
+
+  def build_partner(partner_params=nil)
+    @partner = Partner.new(permitted_params(partner_params))
+  end
+
+  def load_partner(id)
+    @partner = Partner.find(id)
+  end
+
+  def update_partner(id, partner_params)
+    self.load_partner(id)
+    @partner.attributes = permitted_params(partner_params)
+  end
+
+  def provinces
+    @provinces ||= Province.all
+  end
+
+  def statuses
+    @statuses ||= Status.all
+  end
+
+private
+
+  def permitted_params(partner_params)
+    if partner_params
+      partner_params.permit(:name, :region, :contact_details, :province_id,
+                            :status_id, :start_date)
+    else
+      {}
```

Comment:
I see what you mean but I think it's better as it is. It will be harder to read the code if we move the params parsing inside the facade and the just call:
- `@partner_facade.build_partner(params)`  instead of `@partner_facade.build_partner(params[:partner])`  **or**
- `@partner_facade.load_partner(params)` instead of `@partner_facade.load_partner(params[:id])`

Not sure. Open for discussion.